src/base/c_utils_posix.h

```diff
@@ -75,28 +75,33 @@ inline Literal GetHomeDir(String* error) {
   return Literal(pw->pw_dir);
 }
 
-inline String GetSelfPath(String* error) {
+inline bool GetSelfPath(String& result, String* error) {
   // TODO: self-path is not a subject to change during the execution. Cache it.
   char path[PATH_MAX];
 #if defined(OS_LINUX)
-  ssize_t read = readlink("/proc/self/exe", path, PATH_MAX);
+  ssize_t read = readlink("/proc/self/exe", path, sizeof(path));
   if (read == -1) {
     GetLastError(error);
-    return String();
+    return false;
   }
   path[read] = '\0';
 #elif defined(OS_MACOSX)
   ui32 size = sizeof(path);
   if (_NSGetExecutablePath(path, &size) == -1) {
     // TODO: handle not-enough-sized buffer issue.
```

Comment:
I thought the real issue is about resizing the buffer (according to the value written to `size` by `_NSGetExecutablePath`) so we can get the path thereafter, and this issue is not resolved yet. Am I wrong?